core/types/transaction.go

```diff
@@ -128,6 +128,11 @@ func (tx *Transaction) EncodeRLP(w io.Writer) error {
 
 // encodeTyped writes the canonical encoding of a typed transaction to w.
 func (tx *Transaction) encodeTyped(w *bytes.Buffer) error {
+	if tx.Type() == ArbitrumSubtypedTxType {
+		w.WriteByte(tx.Type())
+		w.WriteByte(tx.inner.(*ArbitrumSubtypedTx).TxSubtype())
+		return rlp.Encode(w, tx.inner.(*ArbitrumSubtypedTx).TxData)
+	}
```

Comment:
we need this change here to be able to decode arbitrary internal types that implement TxData (similarly as it is done for top level `Transaction`)